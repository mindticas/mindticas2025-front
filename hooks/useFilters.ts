
import { useState, useCallback } from 'react';

export interface FilterOptions {
    name: string;
    treatments?: string;
    date?: string;
    status?: string;
    [key: string]: string | undefined; // Allow additional properties
  }

export const useFilters = (initialFilters: FilterOptions) => {
  const [filters, setFilters] = useState<FilterOptions>(initialFilters);

  const handleFilterChange = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => {
      const { name, value } = e.target;
      setFilters((prev) => ({ ...prev, [name]: value }));
    },
    []
  );

  const handleStatusChange = useCallback((status: string) => {
    setFilters((prev) => ({ ...prev, status }));
  }, []);

  const resetFilters = useCallback(() => {
    setFilters(initialFilters);
  }, [initialFilters]);

  return {
    filters,
    handleFilterChange,
    handleStatusChange,
    resetFilters,
    setFilters,
  };
};