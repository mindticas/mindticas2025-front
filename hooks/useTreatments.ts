import { Treatment } from '@/interfaces/treatment/Treatment';
import { getTreatments } from '@/services/TreatmentService';
import { useCallback, useEffect, useState } from 'react';

export const useTreatments = (sort?: string) => {
    const [treatments, setTreatments] = useState<Treatment[]>([]);
    const [loading, setLoading] = useState(true);
    const [error, setError] = useState<string | null>(null);

    const fetchTreatments = useCallback(async () => {
        setLoading(true);
        try {
            const data = await getTreatments(sort);
            setTreatments(data);
            setError(null);
        } catch (error) {
            setError(
                'No se pudieron cargar los tratamientos. IntÃ©ntalo de nuevo',
            );
        } finally {
            setLoading(false);
        }
    }, [sort]);

    useEffect(() => {
        fetchTreatments();
    }, [fetchTreatments]);
    return { treatments, loading, error, refetch: fetchTreatments };
};
