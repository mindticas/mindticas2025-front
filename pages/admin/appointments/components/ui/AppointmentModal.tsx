'use client';

import { useEffect, useState } from 'react';
import {
    Dialog,
    Portal,
    Box,
    Input,
    Button,
    Text,
    Field,
    Spinner,
    SelectRoot,
    SelectContent,
    SelectItem,
    SelectTrigger,
    SelectLabel,
    CloseButton,
} from '@chakra-ui/react';
import { Treatment } from '@/interfaces/treatment/Treatment';
import { createListCollection } from '@chakra-ui/react';

interface AppointmentModalProps {
    isOpen: boolean;
    onClose: () => void;
    mode: 'create' | 'edit' | 'cancel';
    treatments: Treatment[];
    initialData?: {
        id?: number;
        name: string;
        treatment: string;
        date: string;
        time: string;
    };
    onConfirm: (data: {
        id?: number;
        name: string;
        treatment: string;
        date: string;
        time: string;
    }) => Promise<void>;
    onCancel?: (id: number) => Promise<void>;
}

export default function AppointmentModal({
    isOpen,
    onClose,
    mode,
    treatments,
    initialData = {
        name: '',
        treatment: '',
        date: '',
        time: '',
    },
    onConfirm,
    onCancel,
}: AppointmentModalProps) {
    const [formData, setFormData] = useState({
        name: initialData.name,
        treatment: initialData.treatment,
        date: initialData.date,
        time: initialData.time,
    });

    useEffect(() => {
        if (isOpen) {
            setFormData({
                name: initialData.name,
                treatment: initialData.treatment,
                date: initialData.date,
                time: initialData.time,
            });
        }
    }, [
        isOpen,
        initialData.id,
        initialData.name,
        initialData.treatment,
        initialData.date,
        initialData.time,
    ]);

    const [isLoading, setIsLoading] = useState(false);

    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const { name, value } = e.target;
        setFormData((prev) => ({ ...prev, [name]: value }));
    };

    const handleSubmit = async (e: React.FormEvent) => {
        e.preventDefault();
        setIsLoading(true);
        try {
            await onConfirm({
                id: initialData?.id,
                ...formData,
            });
            onClose();
        } finally {
            setIsLoading(false);
        }
    };

    const handleCancelAppointment = async () => {
        if (mode === 'cancel' && initialData?.id && onCancel) {
            setIsLoading(true);
            try {
                await onCancel(initialData.id);
                onClose();
            } finally {
                setIsLoading(false);
            }
        }
    };

    return (
        <Dialog.Root
            open={isOpen}
            onOpenChange={({ open }) => !open && onClose()}
            placement='center'
        >
            <Portal>
                <Dialog.Backdrop />
                <Dialog.Positioner>
                    <Dialog.Content p={5} backgroundColor='white'>
                        <Dialog.Header
                            mb={4}
                            display='flex'
                            justifyContent='space-between'
                            alignContent='center'
                        >
                            <Dialog.Title>
                                {mode === 'cancel'
                                    ? 'Confirmar cancelaci√≥n'
                                    : mode === 'edit'
                                    ? 'Editar Cita'
                                    : 'Nueva Cita'}
                            </Dialog.Title>
                            <Dialog.CloseTrigger asChild>
                                <CloseButton color='black' size='sm' />
                            </Dialog.CloseTrigger>
                        </Dialog.Header>

                        <Dialog.Body>
                            {mode !== 'cancel' ? (
                                <form onSubmit={handleSubmit}>
                                    <Box mb={4}>
                                        <Field.Root>
                                            <Field.Label fontWeight='semibold'>
                                                Nombre
                                            </Field.Label>
                                            <Input
                                                p={2}
                                                name='name'
                                                placeholder='Nombre del Cliente'
                                                value={formData.name}
                                                onChange={handleInputChange}
                                                required
                                            />
                                        </Field.Root>
                                    </Box>

                                    <Box mb={4}>
                                        {treatments && (
                                            <SelectRoot
                                                name='treatment'
                                                collection={createListCollection(
                                                    {
                                                        items: treatments,
                                                    },
                                                )}
                                                onValueChange={(e) => {
                                                    const selectedTreatment =
                                                        treatments.find(
                                                            (treatment) =>
                                                                treatment.name ===
                                                                e.value.toString(),
                                                        );
                                                    setFormData({
                                                        ...formData,
                                                        treatment:
                                                            selectedTreatment
                                                                ? selectedTreatment.id.toString()
                                                                : '',
                                                    });
                                                }}
                                                required
                                            >
                                                <SelectLabel fontWeight='semibold'>
                                                    Servicio a realizar
                                                </SelectLabel>
                                                <SelectTrigger>
                                                    {(treatments &&
                                                        treatments.find(
                                                            (t) =>
                                                                t.id.toString() ===
                                                                formData.treatment,
                                                        )?.name) ||
                                                        'Seleccionar servicio'}
                                                </SelectTrigger>
                                                <SelectContent
                                                    className='select-content-admin-edit'
                                                    backgroundColor='white'
                                                >
                                                    {treatments &&
                                                        treatments.map(
                                                            ({ id, name }) => (
                                                                <SelectItem
                                                                    cursor='pointer'
                                                                    _hover={{
                                                                        backgroundColor:
                                                                            'gray.100',
                                                                    }}
                                                                    backgroundColor='white'
                                                                    item={name}
                                                                    key={id}
                                                                    p={2}
                                                                    data-state={
                                                                        id ===
                                                                        id
                                                                            ? 'checked'
                                                                            : 'unchecked'
                                                                    }
                                                                >
                                                                    {name}
                                                                </SelectItem>
                                                            ),
                                                        )}
                                                </SelectContent>
                                            </SelectRoot>
                                        )}
                                    </Box>

                                    <Box mb={4}>
                                        <Field.Root>
                                            <Field.Label fontWeight='semibold'>
                                                Fecha
                                            </Field.Label>
                                            <Input
                                                _dark={{
                                                    '&::-webkit-calendar-picker-indicator':
                                                        {
                                                            filter: 'invert(1)',
                                                        },
                                                }}
                                                p={2}
                                                type='date'
                                                name='date'
                                                value={formData.date}
                                                onChange={handleInputChange}
                                                required
                                            />
                                        </Field.Root>
                                    </Box>

                                    <Box mb={4}>
                                        <Field.Root>
                                            <Field.Label fontWeight='semibold'>
                                                Hora
                                            </Field.Label>
                                            <Input
                                                _dark={{
                                                    '&::-webkit-calendar-picker-indicator':
                                                        {
                                                            filter: 'invert(1)',
                                                        },
                                                }}
                                                p={2}
                                                type='time'
                                                name='time'
                                                value={formData.time}
                                                onChange={handleInputChange}
                                                required
                                            />
                                        </Field.Root>
                                    </Box>
                                </form>
                            ) : (
                                <Text>
                                    ¬øEst√°s seguro que deseas cancelar esta cita?
                                </Text>
                            )}
                        </Dialog.Body>

                        <Dialog.Footer p={4}>
                            <Button
                                backgroundColor={
                                    mode === 'cancel' ? 'black' : 'red'
                                }
                                color='white'
                                p={3}
                                onClick={onClose}
                            >
                                {mode === 'cancel' ? 'Volver' : 'Cancelar'}
                            </Button>

                            <Button
                                colorScheme={mode === 'cancel' ? 'red' : 'blue'}
                                ml={3}
                                onClick={
                                    mode === 'cancel'
                                        ? handleCancelAppointment
                                        : handleSubmit
                                }
                                p={3}
                                backgroundColor={
                                    mode === 'cancel' ? 'red' : '#1C4ED8'
                                }
                                color='white'
                                disabled={isLoading}
                            >
                                {isLoading ? (
                                    <Spinner size='sm' mr={2} />
                                ) : mode === 'cancel' ? (
                                    'Confirmar cancelaci√≥n'
                                ) : mode === 'edit' ? (
                                    'Actualizar'
                                ) : (
                                    'Guardar'
                                )}
                            </Button>
                        </Dialog.Footer>
                    </Dialog.Content>
                </Dialog.Positioner>
            </Portal>
        </Dialog.Root>
    );
}
