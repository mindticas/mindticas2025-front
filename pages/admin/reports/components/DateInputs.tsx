import { StatisticsData } from '@/interfaces/statistics/StatisticsData';
import { StatisticsDataResponse } from '@/interfaces/statistics/StatisticsDataResponse';
import { Treatment } from '@/interfaces/treatment/Treatment';
import {
    getStatistics,
    exportStatisticsToExcel,
} from '@/services/StatisticsService';
import {
    Text,
    Box,
    Flex,
    Input,
    SelectRoot,
    SelectTrigger,
    SelectHiddenSelect,
    SelectControl,
    SelectIndicator,
    SelectIndicatorGroup,
    Portal,
    SelectPositioner,
    SelectValueText,
    SelectItemIndicator,
    useBreakpointValue,
} from '@chakra-ui/react';
import { createListCollection } from '@chakra-ui/react';
import { SelectLabel, SelectContent, SelectItem } from '@chakra-ui/react';
import { Download } from 'lucide-react';
import { useEffect, useState } from 'react';

interface DateInputsProps {
    onDateChange: (dates: StatisticsData) => void;
    onStatisticsFetched: (data: StatisticsDataResponse | null) => void;
    onTreatmentSelect: (treatmentId: string) => void;
    onLoadingChange: (isLoading: boolean) => void;
    treatments: Treatment[];
    treatmentsError: string | null;
}

export default function DateInputs({
    onStatisticsFetched,
    onDateChange,
    onTreatmentSelect,
    onLoadingChange,
    treatments,
    treatmentsError,
}: DateInputsProps) {
    const [formData, setFormData] = useState<StatisticsData>({
        startDate: '',
        endDate: '',
        treatment: '',
    });
    const [selectTreatment, setSelectTreatment] = useState('');
    const [statisticsLoading, setStatisticsLoading] = useState(false);
    const [statisticsError, setStatisticsError] = useState<string | null>(null);
    const isMobile = useBreakpointValue({
        base: true,
        sm: true,
        md: false,
        lg: false,
    });

    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const { name, value } = e.target;
        setFormData((prev) => ({
            ...prev,
            [name]: value,
        }));
    };

    const handleSelectChange = (value: string) => {
        setSelectTreatment(value.toString());
        setFormData((prev) => ({
            ...prev,
            treatment: value.toString(),
        }));
        onTreatmentSelect(value);
    };

    useEffect(() => {
        if (!formData.startDate || !formData.endDate) {
            onDateChange({ startDate: '', endDate: '' });
            onStatisticsFetched(null);
            return;
        }
        onDateChange({
            startDate: formData.startDate,
            endDate: formData.endDate,
        });
        fetchStatistics();
    }, [formData]);

    useEffect(() => {
        if (!formData.startDate || !formData.endDate) {
        }
    });

    useEffect(() => {
        onLoadingChange(statisticsLoading);
    }, [statisticsLoading, onLoadingChange]);

    const fetchStatistics = async () => {
        // Verify that the dates have been selected
        if (!formData.startDate || !formData.endDate) {
            setStatisticsError('Selecciona fechas de inicio y fin');
            return;
        }
        setStatisticsLoading(true);
        setStatisticsError(null);
        try {
            const data = await getStatistics(formData);
            // Send data to index
            onStatisticsFetched(data);
        } catch (error) {
            setStatisticsError('Error al cargar las estadisticas');
        } finally {
            setStatisticsLoading(false);
        }
    };

    const handleClearTreatment = () => {
        setSelectTreatment('');
        setFormData((prev) => ({ ...prev, treatment: '' }));
        onTreatmentSelect('');
    };

    const handleDownloadExcel = async () => {
        if (!formData.startDate || !formData.endDate) {
            setStatisticsError('Selecciona fechas de inicio y fin');
            return;
        }
        if (formData.treatment) {
            setStatisticsError(
                'La exportación solo está disponible con fechas. Intente nuevamente sin seleccionar un servicio',
            );
            return;
        }
        try {
            setStatisticsLoading(true);
            await exportStatisticsToExcel(formData);
        } catch (error) {
            setStatisticsError('Error al descargar el archivo');
        } finally {
            setStatisticsLoading(false);
        }
    };

    return (
        <Box px={{ base: 4, sm: 6, md: 8, lg: 10 }} py={6}>
            <Flex
                mt='10'
                gap={{ base: '4', md: '0' }}
                justifyContent='space-between'
                direction={{ base: 'column', md: 'row' }}
                align={{ base: 'flex-start', md: 'center' }}
            >
                <Text fontSize='3xl' fontWeight='bold'>
                    Reportes
                </Text>

                <Flex
                    gap={['2', '4']}
                    direction={['column', 'column', 'row', 'row']}
                    width={['full', 'full', 'auto', 'auto']}
                >
                    <Box>
                        <Text fontSize='md' fontWeight='semibold'>
                            Fecha inicial
                        </Text>
                        <Input
                            p={2}
                            type='date'
                            name='startDate'
                            size='lg'
                            _dark={{
                                '&::-webkit-calendar-picker-indicator': {
                                    filter: 'invert(1)',
                                },
                            }}
                            value={formData.startDate}
                            onChange={handleInputChange}
                        />
                    </Box>
                    <Box>
                        <Text fontSize='md' fontWeight='semibold'>
                            Fecha final
                        </Text>
                        <Input
                            p={2}
                            size='lg'
                            type='date'
                            name='endDate'
                            _dark={{
                                '&::-webkit-calendar-picker-indicator': {
                                    filter: 'invert(1)',
                                },
                            }}
                            value={formData.endDate}
                            onChange={handleInputChange}
                        />
                    </Box>
                    <Box>
                        <SelectRoot
                            collection={createListCollection({
                                items: Array.isArray(treatments)
                                    ? treatments
                                    : [],
                            })}
                            size='md'
                            onValueChange={(e) => {
                                handleSelectChange(e.value.toString());
                            }}
                            name='treatment'
                            value={[selectTreatment]}
                        >
                            <SelectHiddenSelect />
                            <SelectLabel>
                                <Text fontWeight='semibold' fontSize='md'>
                                    Selecciona un servicio
                                </Text>
                            </SelectLabel>
                            <SelectControl bg='white'>
                                <SelectTrigger>
                                    {selectTreatment || 'Seleccionar servicio'}
                                    <SelectValueText />
                                </SelectTrigger>
                                <SelectIndicatorGroup>
                                    <SelectIndicator />
                                </SelectIndicatorGroup>
                            </SelectControl>
                            <Portal>
                                <SelectPositioner>
                                    <Box>
                                        <SelectContent
                                            className='select-content-service'
                                            bg='white'
                                            boxShadow='sm'
                                        >
                                            <SelectItem
                                                key='none'
                                                item=''
                                                onClick={() =>
                                                    handleClearTreatment()
                                                }
                                                borderBlockEnd='1px solid'
                                                borderBlockEndColor='gray.200'
                                            >
                                                Sin servicio
                                            </SelectItem>
                                            {treatments &&
                                                treatments.map(
                                                    ({ id, name }) => (
                                                        <SelectItem
                                                            cursor='pointer'
                                                            borderBlockEnd='1px solid'
                                                            borderBlockEndColor='gray.200'
                                                            key={id}
                                                            item={name}
                                                            onChange={
                                                                handleInputChange
                                                            }
                                                        >
                                                            {name}
                                                            <SelectItemIndicator />
                                                        </SelectItem>
                                                    ),
                                                )}
                                        </SelectContent>
                                    </Box>
                                </SelectPositioner>
                            </Portal>
                        </SelectRoot>
                    </Box>
                    {isMobile ? (
                        <Flex
                            gap={2}
                            mt={2}
                            border='1px solid'
                            p='2'
                            borderRadius='sm'
                            justifyContent='space-between'
                            cursor='pointer'
                            fontSize='md'
                            onClick={handleDownloadExcel}
                        >
                            <Text>Descargar excel</Text>
                            <Download />
                        </Flex>
                    ) : (
                        <Box position='relative' p={2} mt='6'>
                            <Download
                                cursor='pointer'
                                onClick={handleDownloadExcel}
                            />
                        </Box>
                    )}
                </Flex>
            </Flex>
            <Box textAlign='center' mx={['4', '8', '12', '14']} mt='6'>
                {treatmentsError && (
                    <Text color='red.500'>{treatmentsError}</Text>
                )}
                {statisticsError && (
                    <Text color='red.500' mt={treatmentsError ? 2 : 0}>
                        {statisticsError}
                    </Text>
                )}
            </Box>
        </Box>
    );
}
