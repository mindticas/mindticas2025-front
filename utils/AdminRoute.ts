import Layout from '@/components/Layout';
import { Box } from '@chakra-ui/react';

export class AdminRoute {
    // Fixed route for admin
    private static route: string = '/admin';

    // Returns the Layout component if the pathname contains '/admin', otherwise returns box
    static getLayout(pathname: string) {
        return pathname.includes(this.route) ? Layout : Box;
    }
}
