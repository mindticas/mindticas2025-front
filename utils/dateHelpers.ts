// Gets the name of the day according to the full calendar format
export const getDayName = (dateStr: string): string => {
    const [year, month, day] = dateStr.split('-').map(Number)
    const date = new Date(year, month -1, day)
    return date.toLocaleDateString('es-ES', { weekday: 'long'})
}
// Get the milliseconds in a day
export function addDays(date: Date, days: number): Date {
    const milisecondsPerDay = 24 * 60 * 60 * 1000; 
    return new Date(date.getTime() + days * milisecondsPerDay)
}