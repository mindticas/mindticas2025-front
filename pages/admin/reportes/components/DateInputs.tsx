import { useTreatments } from '@/hooks/useTreatments';
import { StatisticsData } from '@/interfaces/statistics/StatisticsData';
import { StatisticsDataResponse } from '@/interfaces/statistics/StatisticsDataResponse';
import { getStatistics } from '@/services/StatisticsService';
import {
    Text,
    Box,
    Flex,
    Input,
    SelectRoot,
    SelectTrigger,
    SelectHiddenSelect,
    SelectControl,
    SelectIndicator,
    SelectIndicatorGroup,
    Portal,
    SelectPositioner,
    SelectValueText,
    SelectItemIndicator,
} from '@chakra-ui/react';

import { createListCollection } from '@chakra-ui/react';
import { SelectLabel, SelectContent, SelectItem } from '@chakra-ui/react';
import { useEffect, useState } from 'react';

interface DateInputsProps {
    onDateChange: (dates: StatisticsData) => void;
    onStatisticsFetched: (data: StatisticsDataResponse) => void;
    onTreatmentSelect: (treatmentId: string) => void;
}

export default function DateInputs({
    onStatisticsFetched,
    onDateChange,
    onTreatmentSelect,
}: DateInputsProps) {
    const { treatments, error } = useTreatments();
    const [formData, setFormData] = useState<StatisticsData>({
        startDate: '',
        endDate: '',
        treatment: '',
    });

    // Colocar un estado para mostrar el nombre del servicio seleccionado
    const [selectTreatment, setSelectTreatment] = useState('');
    const [statistics, setStatistics] = useState<StatisticsDataResponse | null>(
        null,
    );
    const [statisticsLoading, setStatisticsLoading] = useState(false);
    const [statisticsError, setStatisticsError] = useState<string | null>(null);

    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const { name, value } = e.target;
        setFormData((prev) => ({
            ...prev,
            [name]: value,
        }));
    };

    const handleSelectChange = (value: string) => {
        setSelectTreatment(value.toString());
        setFormData((prev) => ({
            ...prev,
            treatment: value.toString(),
        }));
        onTreatmentSelect(value);
    };

    // when dates or treatment change, launch the request
    useEffect(() => {
        if (formData.startDate && formData.endDate) {
            onDateChange?.({
                startDate: formData.startDate,
                endDate: formData.endDate,
            });
            fetchStatistics();
        }
    }, [formData]);

    const fetchStatistics = async () => {
        // Verificar que se hayan seleccionado las fechas
        if (!formData.startDate || !formData.endDate) {
            setStatisticsError('Selecciona fechas de inicio y fin');
            return;
        }
        setStatisticsLoading(true);
        setStatisticsError(null);
        try {
            const data = await getStatistics(formData);
            // send data to index
            onStatisticsFetched(data);
        } catch (error) {
            setStatisticsError('Error al cargar las estadisticas');
        } finally {
            setStatisticsLoading(false);
        }
    };

    return (
        <>
            <Flex
                mx={['4', '8', '12', '14']}
                p='4'
                mt='10'
                gap={['4', '6', '0', '0']}
                justifyContent='space-between'
                direction={['column', 'colum', 'row', 'row']}
                alignItems='center'
            >
                <Text fontSize='3xl' fontWeight='bold'>
                    Reportes
                </Text>

                <Flex
                    gap={['2', '4']}
                    direction={['column', 'column', 'row', 'row']}
                    width={['full', 'full', 'auto', 'auto']}
                >
                    <Box>
                        <Text fontSize='md' fontWeight='semibold'>
                            Fecha inicial
                        </Text>
                        <Input
                            p={2}
                            type='date'
                            name='startDate'
                            size='lg'
                            _dark={{
                                '&::-webkit-calendar-picker-indicator': {
                                    filter: 'invert(1)',
                                },
                            }}
                            onChange={handleInputChange}
                        />
                    </Box>
                    <Box>
                        <Text fontSize='md' fontWeight='semibold'>
                            Fecha final
                        </Text>
                        <Input
                            p={2}
                            size='lg'
                            type='date'
                            name='endDate'
                            _dark={{
                                '&::-webkit-calendar-picker-indicator': {
                                    filter: 'invert(1)',
                                },
                            }}
                            onChange={handleInputChange}
                        />
                    </Box>
                    <Box>
                        <SelectRoot
                            collection={createListCollection({
                                items: treatments,
                            })}
                            size='md'
                            onValueChange={(e) =>
                                handleSelectChange(e.value.toString())
                            }
                            name='treatment'
                        >
                            <SelectHiddenSelect />
                            <SelectLabel>
                                <Text fontWeight='semibold' fontSize='md'>
                                    Selecciona un servicio
                                </Text>
                            </SelectLabel>
                            <SelectControl bg='white'>
                                <SelectTrigger>
                                    {selectTreatment || 'Seleccionar servicio'}
                                    <SelectValueText />
                                </SelectTrigger>
                                <SelectIndicatorGroup>
                                    <SelectIndicator />
                                </SelectIndicatorGroup>
                            </SelectControl>
                            <Portal>
                                <SelectPositioner>
                                    <Box>
                                        <SelectContent
                                            className='select-content-service'
                                            bg='white'
                                            boxShadow='sm'
                                        >
                                            {treatments.map(({ id, name }) => (
                                                <SelectItem
                                                    cursor='pointer'
                                                    borderBlockEnd='1px solid'
                                                    borderBlockEndColor='gray.200'
                                                    key={id}
                                                    item={name}
                                                    onChange={handleInputChange}
                                                >
                                                    {name}
                                                    <SelectItemIndicator />
                                                </SelectItem>
                                            ))}
                                        </SelectContent>
                                    </Box>
                                </SelectPositioner>
                            </Portal>
                        </SelectRoot>
                    </Box>
                </Flex>
            </Flex>
            <Box textAlign='center' mx={['4', '8', '12', '14']} mt='6'>
                {error && <Text color='red.500'>{error}</Text>}
            </Box>
        </>
    );
}
