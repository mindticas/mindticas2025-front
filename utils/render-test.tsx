import { Provider } from '@/components/ui/provider';
import { BookingProvider } from '@/context/BookingContext';
import { render as rtlRender } from '@testing-library/react';

export function render(ui: React.ReactNode) {
    return rtlRender(<>{ui}</>, {
        wrapper: (props: React.PropsWithChildren) => (
            <Provider>
                <BookingProvider>{props.children}</BookingProvider>
            </Provider>
        ),
    });
}
