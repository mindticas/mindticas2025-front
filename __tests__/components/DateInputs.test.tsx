import { Provider } from '@/components/ui/provider';
import DateInputs from '@/pages/admin/reports/components/DateInputs';
import { render } from '@/utils/render-test';

// Mock of hooks and services
jest.mock('@/hooks/useTreatments', () => ({
    useTreatments: () => ({
        treatments: [
            { id: '1', name: 'Tratamiento 1' },
            { id: '2', name: 'Tratamiento 2' },
        ],
        error: null,
    }),
}));

jest.mock('@/services/StatisticsService', () => ({
    getStatistics: jest.fn(),
}));

describe('DateInputs component', () => {
    const mockProps = {
        onDateChange: jest.fn(),
        onStatisticsFetched: jest.fn(),
        onTreatmentSelect: jest.fn(),
        onLoadingChange: jest.fn(),
    };

    it('matches snapshot with default state', () => {
        const { asFragment } = render(
            <Provider>
                <DateInputs {...mockProps} />
            </Provider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot with error', () => {
        // Mock con error
        jest.spyOn(
            require('@/hooks/useTreatments'),
            'useTreatments',
        ).mockImplementation(() => ({
            treatments: [],
            error: 'Error al cargar tratamientos',
        }));

        const { asFragment } = render(
            <Provider>
                <DateInputs {...mockProps} />
            </Provider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
