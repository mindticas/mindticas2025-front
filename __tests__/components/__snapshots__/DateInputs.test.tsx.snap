// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInputs component matches snapshot with default state 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-fcx4l6"
  >
    <p
      class="css-1bzswz8"
    >
      Reportes
    </p>
    <div
      class="css-39rk59"
    >
      <div
        class="css-0"
      >
        <p
          class="css-1hrrvn"
        >
          Fecha inicial
        </p>
        <input
          class="chakra-input css-oyzv26"
          name="startDate"
          type="date"
        />
      </div>
      <div
        class="css-0"
      >
        <p
          class="css-1hrrvn"
        >
          Fecha final
        </p>
        <input
          class="chakra-input css-oyzv26"
          name="endDate"
          type="date"
        />
      </div>
      <div
        class="css-0"
      >
        <div
          class="chakra-select__root css-3jlqt6"
          data-part="root"
          data-scope="select"
          dir="ltr"
          id="select::r0:"
        >
          <select
            aria-hidden="true"
            aria-labelledby="select::r0::label"
            id="select::r0::select"
            name="treatment"
            style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap; word-wrap: normal;"
            tabindex="-1"
          >
            <option
              value=""
            />
            <option
              value=""
            />
            <option
              value=""
            />
          </select>
          <label
            class="chakra-select__label css-p3rj3s"
            data-part="label"
            data-scope="select"
            dir="ltr"
            for="select::r0::select"
            id="select::r0::label"
          >
            <p
              class="css-1mv89is"
            >
              Selecciona un servicio
            </p>
          </label>
          <div
            class="chakra-select__control css-cmjsi3"
            data-part="control"
            data-scope="select"
            data-state="closed"
            dir="ltr"
            id="select::r0::control"
          >
            <button
              aria-controls="select::r0::content"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="select::r0::label"
              class="chakra-select__trigger css-12uc8un"
              data-part="trigger"
              data-placeholder-shown=""
              data-scope="select"
              data-state="closed"
              dir="ltr"
              id="select::r0::trigger"
              role="combobox"
              type="button"
            >
              Seleccionar servicio
              <span
                class="chakra-select__valueText css-176t9sc"
                data-part="value-text"
                data-scope="select"
                dir="ltr"
              />
            </button>
            <div
              class="chakra-select__indicatorGroup css-1xvjckd"
            >
              <div
                aria-hidden="true"
                class="chakra-select__indicator css-1atg1ut"
                data-part="indicator"
                data-scope="select"
                data-state="closed"
                dir="ltr"
              >
                <svg
                  class="css-s3mb0o"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-18utin8"
  />
</DocumentFragment>
`;

exports[`DateInputs component matches snapshot with error 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-fcx4l6"
  >
    <p
      class="css-1bzswz8"
    >
      Reportes
    </p>
    <div
      class="css-39rk59"
    >
      <div
        class="css-0"
      >
        <p
          class="css-1hrrvn"
        >
          Fecha inicial
        </p>
        <input
          class="chakra-input css-oyzv26"
          name="startDate"
          type="date"
        />
      </div>
      <div
        class="css-0"
      >
        <p
          class="css-1hrrvn"
        >
          Fecha final
        </p>
        <input
          class="chakra-input css-oyzv26"
          name="endDate"
          type="date"
        />
      </div>
      <div
        class="css-0"
      >
        <div
          class="chakra-select__root css-3jlqt6"
          data-part="root"
          data-scope="select"
          dir="ltr"
          id="select::r1:"
        >
          <select
            aria-hidden="true"
            aria-labelledby="select::r1::label"
            id="select::r1::select"
            name="treatment"
            style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap; word-wrap: normal;"
            tabindex="-1"
          >
            <option
              value=""
            />
          </select>
          <label
            class="chakra-select__label css-p3rj3s"
            data-part="label"
            data-scope="select"
            dir="ltr"
            for="select::r1::select"
            id="select::r1::label"
          >
            <p
              class="css-1mv89is"
            >
              Selecciona un servicio
            </p>
          </label>
          <div
            class="chakra-select__control css-cmjsi3"
            data-part="control"
            data-scope="select"
            data-state="closed"
            dir="ltr"
            id="select::r1::control"
          >
            <button
              aria-controls="select::r1::content"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="select::r1::label"
              class="chakra-select__trigger css-12uc8un"
              data-part="trigger"
              data-placeholder-shown=""
              data-scope="select"
              data-state="closed"
              dir="ltr"
              id="select::r1::trigger"
              role="combobox"
              type="button"
            >
              Seleccionar servicio
              <span
                class="chakra-select__valueText css-176t9sc"
                data-part="value-text"
                data-scope="select"
                dir="ltr"
              />
            </button>
            <div
              class="chakra-select__indicatorGroup css-1xvjckd"
            >
              <div
                aria-hidden="true"
                class="chakra-select__indicator css-1atg1ut"
                data-part="indicator"
                data-scope="select"
                data-state="closed"
                dir="ltr"
              >
                <svg
                  class="css-s3mb0o"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-18utin8"
  >
    <p
      class="css-1nyyqq1"
    >
      Error al cargar tratamientos
    </p>
  </div>
</DocumentFragment>
`;
